internal/gateway/schema_load.go

```diff
@@ -43,6 +43,9 @@ func loadEndpointSchema(config Config, upstream *upstreamServer) (*schema.Schema
 			if upstreamSchemaFileExists {
 				config.Log.Printf("download failed (will load cached schema version): %v", err)
 			} else {
+				if err.Error() == "unexpected end of JSON input" {
```

Comment:
Not the best way to handle error

Reply:
should I compare the error
```go
if err == io.ErrUnexpectedEOF {
        return nil, errors.Errorf("download failed: graphQL api doesn't exist at %s", upstream.info.URL)
}
```

or I have to wrap ```io.ErrUnexpectedEOF```  with the error.
```go
if err.Error() == "unexpected end of JSON input" {
	return nil, errors.Wrap(io.ErrUnexpectedEOF, "download failed: graphQL api doesn't exist")
}
```

Reply:
You should make this PR mergable. Using error message to detect error is hack

Reply:
@wtrocki  I was able to do it by using ```json.SyntaxError``` in ```encoding/json``` package. It returns the error when json is not valid. 
io was not detecting the error.
```go
if _, ok := err.(*json.SyntaxError); ok {
            return nil, errors.Errorf("download failed: graphQL api doesn't exist at %s", upstream.info.URL)
}
```